gef.py

```diff
@@ -7792,18 +7817,31 @@ def __get_current_block_start_address():
                         if op in extended_registers[exreg]:
                             parameter_set.add(exreg)
 
-        if is_x86_32():
-            nb_argument = len(parameter_set)
-        else:
-            nb_argument = 0
-            for p in parameter_set:
-                nb_argument = max(nb_argument, function_parameters.index(p)+1)
+        nb_argument = -1
+        if function_name.endswith('@plt'):
+            _function_name = function_name.split('@')[0]
+            try:
+                nb_argument = len(libc_args_definitions[_function_name])
+            except:
+                pass
+
+        if nb_argument < 0:
+            if is_x86_32():
```

Comment:
it would be really nice to have a generalized way to get the number of arguments that is either arch agnostic or in the arch class as a method

Reply:
this behaviour is in current gef, i'm not into gef enough to improve this right now i think